src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/AccountsReferenceDateQuestionController.java

```diff
@@ -0,0 +1,127 @@
+    @Autowired
+    private SmallFullServiceImpl smallFullService;
+
+    @Autowired
+    private ApiClientService apiClientService;
+
+    private static final String ACCOUNTS_REFERENCE_DATE_QUESTION = "accountsReferenceDateQuestion";
+
+    @GetMapping
+    public String getAccountsReferenceDateQuestion(@PathVariable String companyNumber,
+                                                   @PathVariable String transactionId,
+                                                   @PathVariable String companyAccountsId,
+                                                   Model model,
+                                                   HttpServletRequest request) {
+
+        CompanyProfileApi companyProfile;
+        SmallFullApi smallFullAccounts;
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        AccountsReferenceDateQuestion accountsReferenceDateQuestion = new AccountsReferenceDateQuestion();
+
+        try {
+            companyProfile = companyService.getCompanyProfile(companyNumber);
+            smallFullAccounts = smallFullService.getSmallFullAccounts(apiClient, transactionId, companyAccountsId);
+        } catch (ServiceException e) {
+                LOGGER.errorRequest(request, e.getMessage(), e);
+                return ERROR_VIEW;
+        }
+
+        accountsReferenceDateQuestion.setPeriodStartOn(companyProfile.getAccounts().getNextAccounts().getPeriodStartOn());
+        accountsReferenceDateQuestion.setPeriodEndOn(companyProfile.getAccounts().getNextAccounts().getPeriodEndOn());
+
+        LocalDate smallFullPeriodEndOn = smallFullAccounts.getNextAccounts().getPeriodEndOn();
+
+        if (! accountsReferenceDateQuestion.getPeriodEndOn().equals(smallFullPeriodEndOn)) {
+            accountsReferenceDateQuestion.setHasConfirmedAccountingReferenceDate(false);
+        } else {
+            setHasConfirmedAccountingReferenceDate(request, accountsReferenceDateQuestion);
+        }
+
+        model.addAttribute(ACCOUNTS_REFERENCE_DATE_QUESTION, accountsReferenceDateQuestion);
+
+        return getTemplateName();
+    }
+
+    @PostMapping
+    public String submitAccountsReferenceDateQuestion(@PathVariable String companyNumber,
+                                                @PathVariable String transactionId,
+                                                @PathVariable String companyAccountsId,
+                                                @ModelAttribute(ACCOUNTS_REFERENCE_DATE_QUESTION) @Valid AccountsReferenceDateQuestion accountsReferenceDateQuestion,
+                                                BindingResult bindingResult,
+                                                HttpServletRequest request) {
+
+        if (bindingResult.hasErrors()) {
+            return getTemplateName();
+        }
+
+        if (accountsReferenceDateQuestion.getHasConfirmedAccountingReferenceDate()) {
+            try {
+                smallFullService.updateSmallFullAccounts(null, transactionId, companyAccountsId );
```

Comment:
As above but it really is a matter of styling here - there's no performance gain, so ultra nitpick!

Reply:
moved into try block, looks clean so I'm also happy with re-structure